crates/runtime/benches/bench_search/setup.rs

```diff
@@ -58,20 +105,59 @@ async fn build_bench_app(
             item_ttl: None,
             eviction_policy: None,
         })
-        .with_embedding(Embeddings::new(embeddings_model, "test_model"));
+        .with_embedding(create_embeddings_model(embeddings_model));
 
-    add_benchmark_dataset(app_builder, test_dataset, acceleration).await
+    add_benchmark_dataset(app_builder, test_dataset, acceleration.clone()).await
 }
 
 async fn add_benchmark_dataset(
     app_builder: AppBuilder,
     dataset: &str,
     acceleration: Option<Acceleration>,
 ) -> Result<AppBuilder, String> {
-    match dataset {
-        "QuoraRetrieval" => {
+    match dataset.to_lowercase().as_str() {
+        "quoraretrieval" => {
             super::datasets::add_mtep_quora_retrieval_dataset(app_builder, acceleration).await
         }
         _ => Err(format!("Unknown benchmark dataset: {dataset}")),
     }
 }
+
+fn create_embeddings_model(embeddings_model: &str) -> Embeddings {
```

Comment:
between `/benches` and `/tests` I wonder if it's worth moving this spice component declaration to YAML definitions, i.e.
```rust
Embedding::from_str(r#"
 - name: foo
   from: openai:something
  params:
    - endpoint: localhost
"#);

Reply:
Will improve - thank you